configuration/NI.Analysis.targets

```diff
@@ -0,0 +1,110 @@
+    <Analyzer Include="$(NuGetPackagesDirectory)\ni.codequality.analyzers\1.0.2\lib\netstandard1.3\NI.CodeQuality.Analyzers.dll"/>
+    <Analyzer Include="$(NuGetPackagesDirectory)\ni.codequality.analyzers\1.0.2\lib\netstandard1.3\Utilities.dll"/>
+    <Analyzer Include="$(NuGetPackagesDirectory)\nationalinstruments.tools.analyzers.naming\1.0.2\lib\netstandard1.3\NationalInstruments.Tools.Analyzers.Naming.dll"/>
+    <Analyzer Include="$(NuGetPackagesDirectory)\nationalinstruments.tools.analyzers.naming\1.0.2\lib\netstandard1.3\NationalInstruments.Tools.Analyzers.Utilities.dll"/>
+
+    <!-- Configuration files for our analyzers -->
+    <!-- By adding the Link attribute, these files will not be shown in the Solution Explorer for .NET Core projects. -->
+    <AdditionalFiles Include="$(NI1006_BannedMethods)" Condition="Exists('$(NI1006_BannedMethods)')">
+      <Link>$(NI1006_BannedMethods)</Link>
+    </AdditionalFiles>
+    <AdditionalFiles Include="$(NI1704_AdditionalSpellingDictionary)" Condition="Exists('$(NI1704_AdditionalSpellingDictionary)')">
+      <Link>$(CodeAnalysisRuleSet)</Link>
+    </AdditionalFiles>
+    <AdditionalFiles Include="$(CodeAnalysisRuleSet)" Condition="Exists('$(CodeAnalysisRuleSet)')">
+     <Link>$(CodeAnalysisRuleSet)</Link>
+    </AdditionalFiles>
+
+    <AdditionalFiles Include="$(NI1704_GlobalExceptions)" Condition="Exists('$(NI1704_GlobalExceptions)')" />
+  </ItemGroup>
+
+  <!-- By default, the NI.ruleset reports violations as warnings instead of errors. 
+       Set TreatWarningsAsErrors to true in release to generate errors. -->
+  <PropertyGroup Condition="'$(Configuration)' == 'Release'">
+    <TreatWarningsAsErrors>True</TreatWarningsAsErrors>
+  </PropertyGroup>
+
+  <!-- Disable portions of Roslyn analyzers that would otherwise generate errors in our build. There should be a goal
+       around removing as many of these as possible, but we want to report warnings and elevate warnings as errors in
+       our release builds rather than erroring all the time on everything. -->
+  <ItemGroup>
+    <RoslynTransition_DisabledRule Include="0419" />
+    <RoslynTransition_DisabledRule Include="1574" />
+    <RoslynTransition_DisabledRule Include="1580" />
+    <RoslynTransition_DisabledRule Include="1587" />
+    <RoslynTransition_DisabledRule Include="1589" />
+    <RoslynTransition_DisabledRule Include="1658" />
+    <RoslynTransition_DisabledRule Include="1712" />
+    <RoslynTransition_DisabledRule Include="1998" />
+  </ItemGroup>
+  <PropertyGroup>
+    <TemporaryWarningsNotAsErrors Condition="'$(TemporaryWarningsNotAsErrors)' == ''"></TemporaryWarningsNotAsErrors>
+    <WarningsNotAsErrors>$(WarningsNotAsErrors),@(RoslynTransition_DisabledRule),$(TemporaryWarningsNotAsErrors)</WarningsNotAsErrors>
+    <NoWarn>$(NoWarn),@(RoslynTransition_DisabledRule)</NoWarn>
+  </PropertyGroup>
+  <ItemGroup>
+    <HashFiles Include="$(MSBuildThisFileFullPath)" />
```

Comment:
ASW uses this for cache invalidation. Rather than assume all projects do, I think we should remove this and let ASW add e.g. `.nuget\packages.config` to HashFiles

Reply:
I've removed the hash item. We'll need to remember to update that in ASW though at some point